addons/website_sale/static/src/js/checkout.js

```diff
@@ -485,7 +485,9 @@ publicWidget.registry.WebsiteSaleCheckout = publicWidget.Widget.extend({
         const editPickupLocationButton = pickupLocation.querySelector(
             'span[name="o_pickup_location_selector"]'
         );
-        if (editPickupLocationButton.dataset.pickupLocationData) {
+        if (editPickupLocationButton.dataset.pickupLocationData &&
+            Object.keys(JSON.parse(editPickupLocationButton.dataset.pickupLocationData)).length != 0
```

Comment:
```suggestion
        const pickupLocationData = editPickupLocationButton.dataset.pickupLocationData;
        if (pickupLocationData && pickupLocationData !== "{}")
```

since there are some places where we checked "{}" instead converting it in object and then checking object is empty or not :)